x/posts/internal/keeper/handler.go

```diff
@@ -35,6 +35,35 @@ func NewHandler(keeper Keeper) sdk.Handler {
 	}
 }
 
+// ValidatePost checks if the given post is valid according to the current posts' module params
+func ValidatePost(ctx sdk.Context, k Keeper, post types.Post) error {
+	params := k.GetParams(ctx)
+	maxMsgLen := params.MaxPostMessageLength.Int64()
+	maxOpFieldNum := params.MaxOptionalDataFieldsNumber.Int64()
+	maxOpFieldValLen := params.MaxOptionalDataFieldValueLength.Int64()
+
+	if len(post.Message) > int(maxMsgLen) {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+			fmt.Sprintf("Post message cannot exceed %d characters", maxMsgLen))
+	}
+
+	if len(post.OptionalData) > int(maxOpFieldNum) {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+			fmt.Sprintf("Post optional data cannot contain more than %d key-value pairs",
+				maxOpFieldNum))
+	}
+
+	for key, value := range post.OptionalData {
+		if len(value) > int(maxOpFieldValLen) {
+			return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+				fmt.Sprintf("post optional data values cannot exceed %d characters. %s of post with id %s is longer than this",
+					maxOpFieldValLen, key, post.PostID))
+		}
+	}
+
+	return nil
+}
+
```

Comment:
Since this is also used inside the `invariants.go` file, I would move this to a standalone file to make it stand out more